### PR TITLE
feat: add metadata-routed multi-teacher OPD

### PR DESCRIPTION
**Pull request by RyuKosei**
## Summary

  - add metadata-routed multi-teacher on-policy distillation
  - rotate frozen Megatron teachers serially inside the training actor
  - score only each teacher's routed subbatch and restore the student before optimization
  - support mixed datasets and per-sample rollout dispatch
  - reuse Slime's existing sampled-token OPD loss

  ## Design

  Teachers are loaded into pinned CPU backups during actor initialization.
  Only one model is active in the shared GPU buffers at a time. Each trajectory
  is routed to one teacher through dataset metadata, and multi-segment siblings
  are required to use the same teacher.

  This does not introduce separate teacher services or modify Slime source code.

  ## Validation

  - real 8-GPU mixed-teacher optimizer-step smoke
  - both routed teacher forwards completed
  - student restore and OPD optimization completed
  - SGLang weight refresh completed
  - focused MOPD regressions passed
  - public baseline comparison introduced no additional test failures

### CODE REVIEW

**Comment by huang3eng** [comment]
@RyuKosei The current implementation reuses Slime's `token_level` OPD, so I suggest splitting the distillation support into two phases.

The first phase can be completed in this PR: support data-routed, multi-teacher `token_level` MOPD. The configuration and documentation should clearly state that the currently supported distillation type is `token_level`. It would also be helpful to reserve a unified interface for future distillation modes and raise an explicit error for unsupported modes instead of silently falling back to `token_level`.

For a token context `h_t`, routed teacher `d`, and sampled rollout token `y_t`, the current token-level reverse-KL estimator is:

```
reverse_kl_token =
    log p(y_t | h_t) - log q_d(y_t | h_t)
```

If `y_t` is sampled from the current student policy `p`, its expectation is the full-vocabulary reverse KL:

```
E_{y_t ~ p}[reverse_kl_token] = D_KL(p || q_d)
```

If the rollout policy `mu` differs from the current student policy `p`, an importance-sampling correction is required:

```
rho(y_t) = p(y_t | h_t) / mu(y_t | h_t)
```

The second phase can add a logits-based `top_k` implementation in a separate PR. Let `S_k` be the teacher's top-k token set:

```
S_k = TopK_k(q_d)
```

The reverse KL can be decomposed into the contribution from the top-k tokens and the contribution from the remaining vocabulary:

```
D_KL(p || q_d) = D_topk + D_rest
```

where:

```
D_topk =
    sum_{v in S_k} p(v) * log(p(v) / q_d(v))
```

and:

```
D_rest =
    sum_{v not in S_k} p(v) * log(p(v) / q_d(v))
```

If only the teacher's top-k token IDs and probabilities are retained, define the remaining probability masses as:

```
P_tail = 1 - sum_{v in S_k} p(v)
Q_tail = 1 - sum_{v in S_k} q_d(v)
```

A memory-efficient tail-bucket approximation is:

```
D_hat_KL^(k) =
    D_topk + P_tail * log(P_tail / Q_tail)
```

The tail term is necessary because `D_topk` alone does not cover a normalized probability distribution and can even be negative. It also prevents the student from assigning a large amount of probability outside the teacher's top-k set without being penalized.

The tail bucket only captures the total remaining probability mass, so it remains an approximation of the exact remaining-vocabulary contribution. When `k = vocab_size`, the tail is empty and the same formulation reduces to the exact full-vocabulary reverse KL.
